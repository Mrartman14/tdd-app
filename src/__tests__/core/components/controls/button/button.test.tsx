import React from 'react';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { act } from "react-dom/test-utils";

import { Button, ButtonProps } from '../../../../../core/components';

configure({
    adapter: new Adapter(),
});

function setupComponent(props: ButtonProps = {}): ShallowWrapper {
    let component: ShallowWrapper;

    act(() => {
        component = shallow(<Button {...props} />);
    });

    return component!;
}

describe('should button renders well', () => {
    let component: ShallowWrapper;

    afterEach(() => {
        act(() => {
            component.unmount();
        });
    });

    it('should button render children', () => {
        const childrenText = 'children text';
        component = setupComponent({ children: childrenText });
        expect(component.text()).toBe(childrenText);
    });

    // TODO: why this is not passed?
    // it('should button can be disabled', () => {
    //     component = setupComponent({ disabled: true });
    //     expect(component.getElement()).toBeDisabled();
    // });

    it('should onClick prop can be called', () => {
        const mockedOnclick = jest.fn();
        component = setupComponent({ onClick: mockedOnclick });
        expect(mockedOnclick.mock.calls.length).toBe(0);
        act(() => {
            component.simulate('click');
        });
        expect(mockedOnclick.mock.calls.length).toBe(1);
    });

    it('snapshot', () => {
        component = setupComponent({
            disabled: true,
            children: 'button text',
        });
        expect(component).toMatchSnapshot();
    });
});
