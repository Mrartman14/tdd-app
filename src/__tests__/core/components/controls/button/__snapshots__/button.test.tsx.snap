// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should button renders well snapshot 1`] = `
<button
  disabled={true}
>
  button text
</button>
`;
